lambdas/src/apis/profiles/controllers/profiles.ts

```diff
@@ -43,21 +88,31 @@ export async function getProfilesById(
     const profiles = await fetchProfilesForSnapshots(profileIds, client)
     res.send(profiles)
   } else {
-    const profiles = await fetchProfiles(profileIds, client, ensOwnership, wearables)
-    res.send(profiles)
+    const profiles = await fetchProfiles(profileIds, client, ensOwnership, wearables, getIfModifiedSinceTimestamp(req))
+    sendProfilesResponse(res, profiles)
   }
 }
 
+// Dates received from If-Modified-Since headers have precisions of seconds, so we need to round
+function roundToSeconds(timestamp: number) {
+  return Math.floor(timestamp / 1000) * 1000
```

Comment:
are you diving and multiplying for the same number here?

Reply:
We are rounding the timestamp to seconds. So for instance, 1632847801743 will become 1632847801000.

This is because the "If-Modified-Since" header comes with seconds precision: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since. We don't have milliseconds precision there.